### ISSUE TITLE
Hi I am facing this issue while using repair command 

### ISSUE THREAD

**Issue by DnvSaiteja** [SpoonLabs/sorald#335]
java -jar sorald/target/sorald-1.1-SNAPSHOT-jar-with-dependencies.jar repair --original-files-path sorald/src/test/resources/processor_test_files/2095_UnclosedResources/ZipFolder.java --rule-keys 2095 --workspace outputdir

 After executng above command, I am getting the below error

]
INFO  Configured Java source version (sonar.java.source): 14
INFO  JavaClasspath initialization
INFO  JavaClasspath initialization (done) | time=36ms
INFO  JavaTestClasspath initialization
INFO  JavaTestClasspath initialization (done) | time=1ms
INFO  Java Main Files AST scan
INFO  1 source files to be analyzed
INFO  1/1 source files have been analyzed
INFO  Java Main Files AST scan (done) | time=2490ms
INFO  Java Test Files AST scan
INFO  0 source files to be analyzed
INFO  Java Test Files AST scan (done) | time=1ms
INFO  Java Generated Files AST scan
INFO  0/0 source files have been analyzed
INFO  0 source files to be analyzed
INFO  Java Generated Files AST scan (done) | time=0ms
INFO  0/0 source files have been analyzed
java.lang.IllegalArgumentException: 'other' is different type of Path
        at java.base/sun.nio.fs.WindowsPath.relativize(WindowsPath.java:400)
        at java.base/sun.nio.fs.WindowsPath.relativize(WindowsPath.java:42)
        at sorald.event.models.WarningLocation.<init>(WarningLocation.java:16)
        at sorald.event.models.miner.MinedViolationEvent.<init>(MinedViolationEvent.java:24)
        at sorald.Repair.lambda$mineViolations$2(Repair.java:138)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at sorald.Repair.mineViolations(Repair.java:136)
        at sorald.Repair.getRuleViolations(Repair.java:105)
        at sorald.Repair.repair(Repair.java:88)
        at sorald.cli.RepairCommand.call(RepairCommand.java:121)
        at sorald.cli.RepairCommand.call(RepairCommand.java:27)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1933)
        at picocli.CommandLine.access$1200(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2159)
        at picocli.CommandLine.execute(CommandLine.java:2058)
        at sorald.Main.main(Main.java:7)

**Comment by slarse** [comment]
Hi @DnvSaiteja, and thanks for the bug report!

My gut feeling is that this is caused by insufficient Windows support from our side. We don't yet fully support Windows, although that is in the works (#273). It _could_ be something else though, it's hard to say without the files you're trying to repair.

Could you provide the file(s) you are trying to repair? That way, we can try to reproduce the problem.

**Comment by slarse** [comment]
Are you using Windows? It looks like you are as `WindowsPath` is being used, but the paths you're specifying on the command line are Unix paths.

If on Windows, try to change the paths like so:

```bash
java -jar sorald\target\sorald-1.1-SNAPSHOT-jar-with-dependencies.jar repair --original-files-path sorald\src\test\resources\processor_test_files\2095_UnclosedResources\ZipFolder.java --rule-keys 2095 --workspace outputdir
```

**Comment by DnvSaiteja** [comment]
Hi, did anybody found a workaround for how to make sorald work for windows

**Comment by slarse** [comment]
@DnvSaiteja Did you try the things I suggested with the paths? You appear to be supplying Unix-style paths (using `/` as a path separator) on a Windows system (where `\` is the path separator), which probably won't work out.

There are circumstances where Sorald will fail on Windows, and this might be one of those. We are working on Windows support (#273), but it probably won't be ready for another couple of months as we currently have other tasks with higher priority at the moment.

@khaes-kth Can you try to replicate this crash on your system?

**Comment by khaes-kth** [comment]
Hi @slarse 

I just remembered this issue. Sorry for being too late.

I get the same error:

```
INFO  1 source files to be analyzed
INFO  1/1 source files have been analyzed
java.lang.IllegalArgumentException: 'other' is different type of Path
        at java.base/sun.nio.fs.WindowsPath.relativize(WindowsPath.java:400)
        at java.base/sun.nio.fs.WindowsPath.relativize(WindowsPath.java:42)
        at sorald.event.models.WarningLocation.<init>(WarningLocation.java:16)
        at sorald.event.models.miner.MinedViolationEvent.<init>(MinedViolationEvent.java:24)
        at sorald.Repair.lambda$mineViolations$2(Repair.java:138)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at sorald.Repair.mineViolations(Repair.java:136)
        at sorald.Repair.getRuleViolations(Repair.java:105)
        at sorald.Repair.repair(Repair.java:88)
        at sorald.cli.RepairCommand.call(RepairCommand.java:121)
        at sorald.cli.RepairCommand.call(RepairCommand.java:27)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1933)
        at picocli.CommandLine.access$1200(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2159)
        at picocli.CommandLine.execute(CommandLine.java:2058)
        at sorald.Main.main(Main.java:7)
```

The command:
`java -jar sorald.jar repair --original-files-path ..\sorald\src\test\resources\processor_test_files\2095_UnclosedResources\ --rule-keys 2095 --workspace outputdir`

Actually, that is the main issue I always face in Windows. That's why I added #273

**Comment by slarse** [comment]
@khaes-kth thanks for taking a look.

This would then probably be fixed by fixing #166. I will get on that most likely next week as I've faced several difficult-to-debug bugs in the past week simply because of paths being represented by strings a little here and there.
